docs/improving_performance.rst

```diff
@@ -0,0 +1,178 @@
+
+    nengo.Network(seed=1)
+
+2. Disable the operator graph optimizer::
+
+    with nengo.Config(nengo.builder.optimizer.OpMergeOptimizer) as cfg:
+        cfg[nengo.builder.optimizer.OpMergeOptimizer].enabled = False
+        with nengo.Simulator(model) as sim:
+            sim.run(...)
+
+3. Reduce the number of neurons in very large ensembles, or consider using the
+   :class:`nengo.utils.least_squares_solvers.RandomizedSVD` solver.
+
+To improve the simulation run time
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Enable the operator graph optimizer::
+
+    with nengo.Config(nengo.builder.optimizer.OpMergeOptimizer) as cfg:
+        cfg[nengo.builder.optimizer.OpMergeOptimizer].enabled = True
+        with nengo.Simulator(model) as sim:
+            sim.run(...)
+
+2. Consider switching to the `nengo_ocl <https://github.com/nengo/nengo_ocl>`_
+   backend if you have a powerful GPU.
+
+To lower peak memory consumption
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Disable the operator graph optimizer::
+
+    with nengo.Config(nengo.builder.optimizer.OpMergeOptimizer) as cfg:
+        cfg[nengo.builder.optimizer.OpMergeOptimizer].enabled = False
+        with nengo.Simulator(model) as sim:
+            sim.run(...)
+
+2. Reduce the number of neurons in very large ensembles, consider replacing
+   them with multiple smaller ensembles (:class:`nengo.networks.EnsembleArray`
+   is useful for that).
+
+3. Reduce the number of probes or their sampling intervals (with the
+   ``sample_every=`` argument).
+
+
+Build and run time performance
+------------------------------
+
+There are two main things determining how long your model takes to run: the
+build time and the (simulation) run time. The former is the time required to
+translate the model description into the actual neural network that gets
+simulated and is what happens when you create the simulator with
+``nengo.Simulator(model)``. The latter is how long it takes to simulate this
+network for the desired amount of simulation time and is what happens during
+the ``sim.run(...)`` calls.
+
+Some things will influence solely one of these variables, but other things will
+reduce one of the variables at the cost of increasing the other variables.
+Thus, they are a trade-off and the best point therein depends on your exact use
+case.
+
+While the run time is usually the most important aspect, sometimes the memory
+consumption can be the main problem instead.
+
+
+Decoder caching
+---------------
+
+*Influences build time.*
+
+A significant amount of time during the build is spent on finding the NEF
+decoders. Because of that, it is possible to cache the decoders. The first
+build of a model will still take about the same time (technically a bit longer,
+because the computed decoders will be stored), but all subsequent builds of the
+same model can just load the stored decoders and will be faster. This will also
+work to some degree if changes to the model are made, but of course only for
```

Comment:
Remove `of course`. It's kind of condescending. Maybe people don't know what caching is in this context.